sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,286 @@
+		return -1;
+	}
+	result->accels.accelZ = tmp;
+
+	return 0;
+}
+
+
+static int reader_baroDataParse(char *startOfDataSection, time_t timestamp, sensor_event_t *result)
+{
+	char *actField, *nextField;
+	long long tmp;
+
+	actField = startOfDataSection;
+
+	result->type = SENSOR_TYPE_BARO;
+	result->timestamp = timestamp;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->baro.pressure = tmp;
+	actField = nextField;
+
+	if (reader_getFieldLLong(actField, NULL, &tmp) < 0) {
+		return -1;
+	}
+	result->baro.temp = tmp;
+
+	return 0;
+}
+
+
+int reader_read(simsens_reader_t *rd, event_queue_t *queue)
+{
+	char *actField, *nextField;
+	long long tmp;
+	int sensorID, err = 0, emptyIter = 100;
+	ssize_t lineLen;
+	time_t timestamp, timeStart = READER_TIMESTAMP_NOT_SET;
+	sensor_event_t parsed;
+
+	if (rd == NULL || queue == NULL) {
+		return -1;
+	}
+
+	while (!eventQueue_full(queue) && emptyIter > 0) {
+		lineLen = getline(&rd->lineBuf, &rd->bufLen, rd->scenarioFile);
+		if (lineLen == -1) {
+			if (!feof(rd->scenarioFile)) {
+				err = -1;
+				break;
+			}
+
+			clearerr(rd->scenarioFile);
+
+			/* Going to the beginning of the file and omitting file header */
+			if (fseek(rd->scenarioFile, rd->headerLen, SEEK_SET) != 0) {
+				err = -1;
+				break;
+			}
+			emptyIter--;
+
+			continue;
+		}
+
+		if (reader_getFieldLLong(rd->lineBuf, &nextField, &tmp) < 0) {
+			err = -1;
+			break;
+		}
+		sensorID = tmp;
+		actField = nextField;
+
+		/* Checking if we are parsing our header */
+		if ((sensorID & rd->sensorTypes) == 0) {
+			if (sensorID == READER_END_SCENARIO_INDICATOR) {
+				parsed.type = READER_STOP_EVENT;
+				if (eventQueue_enqueue(queue, &parsed) < 0) {
+					err = -1;
+				}
+
+				break;
+			}
+			emptyIter--;
+
+			continue;
+		}
+
+		if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+			err = -1;
+			break;
+		}
+		timestamp = tmp;
+		actField = nextField;
+
+		/* Increasing timeOffset on scenario timestamp decrease (as on file loop) */
+		if (timestamp < rd->timeLast && rd->timeLast != READER_TIMESTAMP_NOT_SET) {
+			rd->timeOffset += rd->timeLast - timestamp;
+		}
+		rd->timeLast = timestamp;
+		timestamp += rd->timeOffset;
+
+		if (timeStart == READER_TIMESTAMP_NOT_SET) {
+			timeStart = timestamp;
+		}
+
+		if (timestamp - timeStart > rd->timeHorizon) {
+			/* Go to start of the line */
+			if (fseek(rd->scenarioFile, -lineLen, SEEK_CUR) != 0) {
+				err = -1;
+			}
+			break;
+		}
+
+		switch (sensorID) {
+			case SENSOR_TYPE_ACCEL:
+				err = reader_accelDataParse(actField, timestamp, &parsed);
+				break;
+			case SENSOR_TYPE_BARO:
+				err = reader_baroDataParse(actField, timestamp, &parsed);
+				break;
+			default:
+				fprintf(stderr, "%s: Unknown sensor type: %d\n", __FUNCTION__, sensorID);
+				err = -1;
+				break;
+		}
+
+		if (err != 0) {
+			break;
+		}
+
+		if (eventQueue_enqueue(queue, &parsed) < 0) {
+			err = -1;
+			break;
+		}
+	}
+
+	return err;
```

Comment:
No check if we exit early because of full queue

Reply:
Because exiting early is not considered as an error